deepchem/models/wandblogger.py

```diff
@@ -0,0 +1,144 @@
+                 resume: Optional[Union[bool, str]] = None,
+                 anonymous: Optional[str] = "never",
+                 log_model: Optional[bool] = False,
+                 **kwargs):
+
+        has_wandb = is_wandb_available()
+        assert has_wandb, "WandbLogger requires wandb to be installed. Run `pip install wandb --upgrade`."
+        if has_wandb:
+            import wandb
+            self.wandb = wandb
+
+        if mode == "offline" and log_model:
+            raise Exception(
+                f'Providing log_model={log_model} and mode={mode} is an invalid configuration'
+                ' since model checkpoints cannot be uploaded in offline mode.\n'
+                'Hint: Set `mode="online"` to log your model.'
+            )
+
+        #Check for datasets and metrics
+        if (train_dataset is None):
+            raise Exception(
+                'No dataset provided: `train_dataset` is not set or is empty. '
+                '`train_dataset` is needed for epoch calculations. '
+            )
+
+        if (metrics is None) or (not metrics):
+            raise Exception(
+                'No metrics are provided: `metrics` is not set or is empty'
+            )
+
+        if eval_logging_strategy != "step" and eval_logging_strategy != "epoch":
+            logger.warning(
+                "Warning: `eval_logging_strategy` needs to be either 'step' or 'epoch'. Defaulting to 'step'."
+            )
+            eval_logging_strategy = "step"
+
+
+        self.datasets = {
+            "train": train_dataset,
+            "eval": eval_dataset
+        }
+        self.train_dataset_size = len(self.datasets["train"])
+        self.metrics = metrics
+        self.eval_logging_strategy = eval_logging_strategy
+
+        self.log_model = log_model
+        self.save_dir = save_dir
+
+        # set wandb init arguments
+        self.wandb_init_params = dict(
+            name=name,
+            project=project,
+            entity=entity,
+            mode=mode,
+            id=id,
+            dir=save_dir,
+            resume=resume,
+            anonymous=anonymous
+        )
+        self.wandb_init_params.update(**kwargs)
+        self.initialized = False
+
+    def setup(self):
+        self.wandb.init(**self.wandb_init_params)
+        self.initialized = True
+
+    def check_other_loggers(self, callbacks):
+        for c in callbacks:
+            if isinstance(c, ValidationCallback):
+                logger.warning(
+                    "Note: You are using both WandbLogger and ValidationCallback. "
+                )
+
+    def calculate_epoch_num(self, current_step):
+        self.steps_per_epoch = math.ceil(self.train_dataset_size / self.wandb.config.batch_size)
+        self.epoch_num = current_step // self.steps_per_epoch
+
+    def log(self, model, extra_data, step):
+
+        all_metrics = dict({})
+
+        #Log Training
+        scores = model.evaluate(self.datasets["train"], self.metrics)
```

Comment:
Lets only log train metrics according to the same criteria as the eval metrics. Otherwise we are running evaluation on the entire training set for every batch.